crates/aptos-rosetta/src/account.rs

```diff
@@ -178,77 +198,111 @@ async fn get_balances(
         },
     };
 
-    // Retrieve staking information (if it applies)
-    // Only non-pool addresses, and non base accounts
+    Ok(sequence_number)
+}
+
+async fn get_staking_info(
+    rest_client: &Client,
+    account: &AccountIdentifier,
+    owner_address: AccountAddress,
+    version: u64,
+) -> ApiResult<(Vec<Amount>, u64, Option<Vec<AccountAddress>>)> {
+    let mut balances = vec![];
+    let mut lockup_expiration: u64 = 0;
     let mut maybe_operators = None;
-    if !account.is_base_account() && pool_address.is_none() {
-        if let Ok(response) = rest_client
-            .get_account_resource_at_version_bcs(
-                owner_address,
-                "0x1::staking_contract::Store",
-                version,
-            )
-            .await
-        {
-            let store: Store = response.into_inner();
-            maybe_operators = Some(vec![]);
-            for (operator, contract) in store.staking_contracts {
-                // Keep track of operators
-                maybe_operators.as_mut().unwrap().push(operator);
-                match get_stake_balances(
-                    rest_client.as_ref(),
+    let mut total_balance = 0;
+    let mut has_staking = false;
+
+    if let Ok(response) = rest_client
+        .get_account_resource_at_version_bcs(owner_address, "0x1::staking_contract::Store", version)
+        .await
+    {
+        let store: Store = response.into_inner();
+        maybe_operators = Some(vec![]);
+        for (operator, contract) in store.staking_contracts {
+            // Keep track of operators
+            maybe_operators.as_mut().unwrap().push(operator);
+            match get_stake_balances(rest_client, account, contract.pool_address, version).await {
+                Ok(Some(balance_result)) => {
+                    if let Some(balance) = balance_result.balance {
+                        has_staking = true;
+                        total_balance += u64::from_str(&balance.value).unwrap_or_default();
```

Comment:
Just to confirm, this isn't a u128?

Reply:
Actually nevermind it will always be a u64